src/components/classrooms/EditClassroomButton.js

```diff
@@ -1,74 +1,186 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import EditIcon from '@material-ui/icons/Create';
+import { withStyles } from '@material-ui/core';
 import { connect } from 'react-redux';
 import Tooltip from '@material-ui/core/Tooltip';
 import Button from '@material-ui/core/Button';
+import Grid from '@material-ui/core/Grid';
+import clsx from 'clsx';
+import CancelIcon from '@material-ui/icons/Cancel';
+import Typography from '@material-ui/core/Typography';
 import IconButton from '@material-ui/core/IconButton';
 import { withTranslation } from 'react-i18next';
-import TextField from '@material-ui/core/TextField';
+import DeleteIcon from '@material-ui/icons/Delete';
 import Dialog from '@material-ui/core/Dialog';
 import DialogActions from '@material-ui/core/DialogActions';
 import DialogContent from '@material-ui/core/DialogContent';
 import DialogTitle from '@material-ui/core/DialogTitle';
 import { editClassroom } from '../../actions';
 import {
   EDIT_CLASSROOM_BUTTON_CLASS,
-  EDIT_CLASSROOM_INPUT_ID,
   EDIT_CLASSROOM_VALIDATE_BUTTON_ID,
   EDIT_CLASSROOM_CANCEL_BUTTON_ID,
+  EDIT_CLASSROOM_DELETE_DATA_BUTTON_CLASS,
 } from '../../config/selectors';
+import ClassroomNameTextField from './ClassroomNameTextField';
+import { isClassroomNameValid } from '../../utils/classroom';
 
-class EditClassroomButton extends Component {
-  state = (() => {
-    const {
-      classroom: { name },
-    } = this.props;
+const styles = theme => ({
+  deleted: {
+    color: 'lightgrey',
+  },
+  dataTitle: {
+    fontSize: 'default',
+    fontWeight: 'bold',
+    marginTop: theme.spacing(2),
+    marginBottom: theme.spacing(1),
+  },
+});
 
-    return {
-      open: false,
-      name,
-    };
-  })();
+const buildDeleteSelection = spaces => {
+  return spaces.reduce((selection, { id: spaceId }) => {
+    return { ...selection, [spaceId]: false };
+  }, {});
+};
 
+class EditClassroomButton extends Component {
   static propTypes = {
+    classes: PropTypes.shape({
+      dataTitle: PropTypes.string.isRequired,
+      deleted: PropTypes.string.isRequired,
+    }).isRequired,
     dispatchEditClassroom: PropTypes.func.isRequired,
     classroom: PropTypes.shape({
       id: PropTypes.string.isRequired,
       name: PropTypes.string.isRequired,
+      spaces: PropTypes.arrayOf({}).isRequired,
     }).isRequired,
     t: PropTypes.func.isRequired,
     userId: PropTypes.string.isRequired,
   };
 
+  state = (() => {
+    const {
+      classroom: { name, spaces },
+    } = this.props;
+
+    // build delete selection on contained spaces for given user
+    const deleteSelection = buildDeleteSelection(spaces);
+
+    return {
+      open: false,
+      name,
+      deleteSelection,
+    };
+  })();
```

Comment:
Do you actually need to call the function with the last `();` bit? I don't remember it like that in the other classes.